frame/staking/src/pallet/impls.rs

```diff
@@ -297,15 +304,19 @@ impl<T: Config> Pallet<T> {
 			RewardDestination::Controller => Self::bonded(stash)
 				.map(|controller| T::Currency::deposit_creating(&controller, amount)),
 			RewardDestination::Stash => T::Currency::deposit_into_existing(stash, amount).ok(),
-			RewardDestination::Staked => Self::bonded(stash)
-				.and_then(|c| Self::ledger(&c).map(|l| (c, l)))
-				.and_then(|(controller, mut l)| {
-					l.active += amount;
-					l.total += amount;
+			RewardDestination::Staked => Self::ledger(Stash(stash.clone()))
+				.and_then(|mut ledger| {
+					ledger.active += amount;
+					ledger.total += amount;
```

Comment:
safer? 
```suggestion
					ledger.active.saturating_accrue(amount);
					ledger.total.saturating_accrue(amount);
```

Reply:
I like these suggestions! I will open a new issue to tackle these in a separate PR. The goal of this PR is to avoid any changes in the staking logic itself out of caution. The changes updates in how the ledger balance is updated are better made in a separate PR.

Reply:
https://github.com/paritytech/substrate/issues/14701